smarts/core/smarts.py

```diff
@@ -330,8 +332,8 @@ def add_provider(self, provider):
         assert isinstance(provider, Provider)
         self._providers.append(provider)
 
-    def switch_ego_agent(self, agent_interface):
-        self._agent_manager.switch_initial_agent(agent_interface)
+    def switch_ego_agents(self, agent_interfaces):
```

Comment:
Consider specifying input/output type hints.

Reply:
Yeah, I considered that here too (after I added them to the `switch_initial_agent()` method), but I rejected that here since `agent_interface` is just being forwarded.  (That is, the type for this function doesn't matter, no methods/fields are accessed and it's passed on regardless, to be checked at the receiver.)